src/txn/txn_rollback_to_stable.c

```diff
@@ -1033,7 +1037,8 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
               __wt_timestamp_to_string(max_durable_ts, ts_string));
 
         /* Cleanup any history store entries for this non-timestamped table. */
-        if (max_durable_ts == WT_TS_NONE && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY)) {
+        if (!S2BT(session)->modified && max_durable_ts == WT_TS_NONE &&
```

Comment:
I don't fully understand this change. Is the reason it's OK because dirty tables get cleaned up via a different rollback to stable mechanism (i.e: the in-memory pass cleans up the data file as well)?

If so I'm uncomfortable, and would like to understand better what goes wrong if modified tables have their history store cleaned up.

Reply:
I added a comment to explain the details when this scenario can happen with an exception, please check and let me know.